### PR TITLE
picom-pijulius: 8.2-unstable-2024-07-01 -> 8.2-unstable-2024-08-30

### PR DESCRIPTION
**Pull request by YvesStraten**
## Description of changes

This PR updates the package, resolving #336787 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TanvirOnGH** [comment]
Fails to build:

```nix
[nix-shell:~/nixpkgs]$ nixpkgs-review pr 338912
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/338912/head:refs/nixpkgs-review/1
remote: Enumerating objects: 46, done.
remote: Counting objects: 100% (42/42), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 46 (delta 20), reused 31 (delta 16), pack-reused 4 (from 1)
Unpacking objects: 100% (46/46), 26.90 KiB | 121.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
 * [new branch]                master                -> refs/nixpkgs-review/0
 * [new ref]                   refs/pull/338912/head -> refs/nixpkgs-review/1
$ git worktree add /home/user/.cache/nixpkgs-review/pr-338912/nixpkgs 2ade30b7af0edc22cc56ea3196b4ab27fa03fb1d
Preparing worktree (detached HEAD 2ade30b7af0e)
Updating files: 100% (43011/43011), done.
HEAD is now at 2ade30b7af0e rustPlatform.buildRustPackage: merge custom platforms
$ git merge --no-commit --no-ff d74c8541033b8605367a855b7f88e356ce2f2fe6
Automatic merge went well; stopped before committing as requested
$ nix build --file /nix/store/zjksks2lpr5y4ibpmlq22f5lh7xwzw6j-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/user/.cache/nixpkgs-review/pr-338912/nixpkgs nixpkgs-overlays=/tmp/tmp8s83m3nn' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/user/.cache/nixpkgs-review/pr-338912/nixpkgs --argstr nixpkgs-config-path /tmp/tmpz0ogqv9a.nix --argstr attrs-path /home/user/.cache/nixpkgs-review/pr-338912/attrs.nix
error: builder for '/nix/store/p8dbzwgc0ndqr8gvvwk2qbx4aq4mg2m2-picom-pijulius-8.2-unstable-2024-08-30.drv' failed with exit code 1;
       last 10 log lines:
       > Run-time dependency libpcre2-8 found: YES 10.44
       > Run-time dependency epoxy found: YES 1.5.10
       > Run-time dependency dbus-1 found: YES 1.14.10
       > Checking for function "dlopen" : YES
       > Compiler for C supports arguments -fsanitize=fuzzer: NO
       > Program asciidoctor found: NO
       >
       > man/meson.build:3:7: ERROR: Program 'asciidoctor' not found or not executable
       >
       > A full log can be found at /build/source/build/meson-logs/meson-log.txt
       For full logs, run 'nix log /nix/store/p8dbzwgc0ndqr8gvvwk2qbx4aq4mg2m2-picom-pijulius-8.2-unstable-2024-08-30.drv'.
error: 1 dependencies of derivation '/nix/store/qm5v4gx1ys7f8hidp885accqsdxfg7dz-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/338912

1 package failed to build:
picom-pijulius
```

**Comment by YvesStraten** [comment]
I'll have to add ASCII doctor then, will do that in a few minutes

**Comment by TanvirOnGH** [comment]
```nix
[nix-shell:~/nixpkgs]$ nixpkgs-review pr 338912
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/338912/head:refs/nixpkgs-review/1
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Total 7 (delta 5), reused 7 (delta 5), pack-reused 0 (from 0)
Unpacking objects: 100% (7/7), 756 bytes | 756.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
   2ade30b7af0e..8899e78e6c2e  master                -> refs/nixpkgs-review/0
 + d74c8541033b...ca37cc729326 refs/pull/338912/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/user/.cache/nixpkgs-review/pr-338912-1/nixpkgs 8899e78e6c2e0a40877525c88389b7f3abf5d408
Preparing worktree (detached HEAD 8899e78e6c2e)
Updating files: 100% (43011/43011), done.
HEAD is now at 8899e78e6c2e cairo-lang: 2.5.4 -> 2.8.0 (#338878)
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/user/.cache/nixpkgs-review/pr-338912-1/nixpkgs nixpkgs-overlays=/tmp/tmpea890ffp -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
error:
       … while querying the derivation named 'sbcl-cl-fam-20121125-git'

       … while calling the 'getAttr' builtin

         at /builtin/derivation.nix:19:19: (source not available)

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'sbcl-cl-fam-20121125-git'
         whose name attribute is located at /home/user/.cache/nixpkgs-review/pr-338912-1/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'nativeLibs' of derivation 'sbcl-cl-fam-20121125-git'

         at /home/user/.cache/nixpkgs-review/pr-338912-1/nixpkgs/pkgs/development/lisp-modules/ql.nix:254:7:

          253|     cl-fam = super.cl-fam.overrideLispAttrs (o: {
          254|       nativeLibs = [ pkgs.fam ];
             |       ^
          255|     });

       error: attribute 'fam' missing

       at /home/user/.cache/nixpkgs-review/pr-338912-1/nixpkgs/pkgs/development/lisp-modules/ql.nix:254:22:

          253|     cl-fam = super.cl-fam.overrideLispAttrs (o: {
          254|       nativeLibs = [ pkgs.fam ];
             |                      ^
          255|     });
       Did you mean one of bam, fac, faq, fasm or fdm?
https://github.com/NixOS/nixpkgs/pull/338912 failed to build: Failed to list packages: nix-env failed with exit code 1
```

**Comment by YvesStraten** [comment]
```
HEAD is now at 8899e78e6c2e cairo-lang: 2.5.4 -> 2.8.0 (#338878)
``` 
Could it be that it is picking the wrong pr?

**Comment by TanvirOnGH** [comment]
Hmm weird.

**Comment by TanvirOnGH** [comment]
Works fine. Tested with pijulius' config. No issues so far.

```nix
Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/338912

1 package built:
picom-pijulius

$ /nix/store/r992f8xgxqsbizyp409hcmpzf72m6j59-nix-2.18.5/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/user/.cache/nixpkgs-review/pr-338912-3/nixpkgs --argstr nixpkgs-config-path /tmp/tmpqcmu0nyq.nix --argstr attrs-path /home/user/.cache/nixpkgs-review/pr-338912-3/attrs.nix --nix-path 'nixpkgs=/home/user/.cache/nixpkgs-review/pr-338912-3/nixpkgs nixpkgs-overlays=/tmp/tmpsegq_4vr' /nix/store/zjksks2lpr5y4ibpmlq22f5lh7xwzw6j-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-338912-3]$ picom
^C[ble: EOF]                                                                                                                                                                            
[nix-shell:~/.cache/nixpkgs-review/pr-338912-3]$ picom --config ~/workspace/dev/picom-pijulius/picom.pijulius.conf
```

**Comment by YvesStraten** [comment]
I successfully built this package on the nixos-vm. I unfortunately cannot give a nixpkgs-review result as it keeps on consistently picking the wrong pr. The binary builds successfully using `nix build .#picom-pijulius`.

**Review by ghost — APPROVED** [review]

File: pkgs/by-name/pi/picom-pijulius/package.nix [hunk 1]
```diff
@@ -4,23 +4,26 @@
...
     owner = "pijulius";
     repo = "picom";
-    rev = "b8fe9323e7606709d692976a7fe7d2455b328bc6";
-    hash = "sha256-bXeoWg1ZukXv+6ZNeRc8gGNsbtBztyW5lpfK0lQK+DE=";
+    rev = "404652dfca5d6708e3a03e78b7e467550a4f7b62";
+    hash = "sha256-VJLMnQpW24OXlCmLoAAkyNMtplzS+NKpUJzLHklkizU=";
   };
 
-  buildInputs = (previousAttrs.buildInputs or [ ]) ++ [ pcre ];
+  buildInputs = (previousAttrs.buildInputs or []) ++ [pcre];
```
Comment (donovanglover) [inline pkgs/by-name/pi/picom-pijulius/package.nix:21]: Please remove the unnecessary formatting changes. These would be overwritten by a nixfmt anyway.

**Review by donovanglover — CHANGES_REQUESTED** [review]
Make sure that the previous version in the commit message is correct. Right now it's missing the year.

File: pkgs/by-name/pi/picom-pijulius/package.nix (hunk 1 above)
Comment (YvesStraten) [reply pkgs/by-name/pi/picom-pijulius/package.nix:21]: alright will do

**Comment by YvesStraten** [comment]
> Make sure that the previous version in the commit message is correct. Right now it's missing the year.

Has been changed, furthermore, the formatting is now fixed. That should wrap up all the comments on this pr. I think that this can be merged now. Thank you very much for the review 🙏

**Review by donovanglover — APPROVED** [review]
Result of `nixpkgs-review pr 338912` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>picom-pijulius</li>
  </ul>
</details>
